src/main/java/net/earthcomputer/clientcommands/command/ItemGroupCommand.java

```diff
@@ -0,0 +1,361 @@
+        if (!groups.containsKey(name)) {
+            throw NOT_FOUND_EXCEPTION.create(name);
+        }
+
+        Group group = groups.get(name);
+        ListTag items = group.getItems();
+        if (index < 0 || index > items.size()) {
+            throw OUT_OF_BOUNDS_EXCEPTION.create(index);
+        }
+        ItemGroup itemGroup = group.getItemGroup();
+        items.remove(index);
+
+        reloadGroups(itemGroup.getIndex());
+        saveFile();
+        sendFeedback("commands.citemgroup.removeStack.success", name, index);
+        return 0;
+    }
+
+    private static int setStack(ServerCommandSource source, String name, int index, ItemStack itemStack) throws CommandSyntaxException {
+        if (!groups.containsKey(name)) {
+            throw NOT_FOUND_EXCEPTION.create(name);
+        }
+
+        Group group = groups.get(name);
+        ListTag items = group.getItems();
+        if ((index < 0) || (index > items.size())) {
+            throw OUT_OF_BOUNDS_EXCEPTION.create(index);
+        }
+        ItemGroup itemGroup = group.getItemGroup();
+        items.set(index, itemStack.toTag(new CompoundTag()));
+
+        reloadGroups(itemGroup.getIndex());
+        saveFile();
+        sendFeedback("commands.citemgroup.setStack.success", name, index, itemStack.getItem().getName());
+        return 0;
+    }
+
+    private static int changeIcon(ServerCommandSource source, String name, ItemStack icon) throws CommandSyntaxException {
+        if (!groups.containsKey(name)) {
+            throw NOT_FOUND_EXCEPTION.create(name);
+        }
+
+        Group group = groups.get(name);
+        ItemGroup itemGroup = group.getItemGroup();
+        ListTag items = group.getItems();
+        ItemStack old = itemGroup.getIcon();
+        itemGroup = FabricItemGroupBuilder.create(
+                new Identifier("clientcommands", name))
+                .icon(() -> icon)
+                .build();
+
+        groups.put(name, new Group(itemGroup, icon, items));
+
+        reloadGroups(group.getItemGroup().getIndex());
+        saveFile();
+        sendFeedback("commands.citemgroup.changeIcon.success", name, old.getItem().getName(), icon.getItem().getName());
+        return 0;
+    }
+
+    private static int renameGroup(ServerCommandSource source, String name, String _new) throws CommandSyntaxException {
+        if (!groups.containsKey(name)) {
+            throw NOT_FOUND_EXCEPTION.create(name);
+        }
+
+        Group group = groups.remove(name);
+        ItemGroup itemGroup = group.getItemGroup();
+        ListTag items = group.getItems();
+        ItemStack icon = itemGroup.getIcon();
+        try {
+            itemGroup = FabricItemGroupBuilder.create(
+                    new Identifier("clientcommands", _new))
+                    .icon(() -> icon)
+                    .build();
+
+            groups.put(_new, new Group(itemGroup, icon, items));
+        } catch (InvalidIdentifierException e) {
```

Comment:
Maybe `Identifier.tryParse` could be used here instead of a try-catch?

Reply:
That returns `null` if it fails, which is not what we want here, do we?

Reply:
Idk I find it cleaner